toolkit/scripts/check_signatures.py

```diff
@@ -20,20 +21,52 @@ def getSignature(fileName) -> str:
             sha256sum.update(read_data)
     return sha256sum.hexdigest()
 
-def check_folder(path):
+def find_file_and_check(path, filename, expected_signature) -> Optional[bool]:
+    path_to_check = os.path.join(path, filename)
+    if Path(path_to_check).is_file():
+        actual_signature = getSignature(path_to_check)
+        if actual_signature == expected_signature:
+            return True
+        print(f"ERROR: detected a mismatched signature for {filename}, expected [{expected_signature}] does not equal actual [{actual_signature}]")
+        return False
+
+    for content in os.listdir(path):
+        path_to_check = os.path.join(path, content)
+        if os.path.isdir(path_to_check):
+            result = find_file_and_check(path_to_check, filename, expected_signature)
+            if result is not None:
+                return result
+
+    return None
+
+def find_spec_folder_with_signatures_json(path: str) -> Optional[str]:
+    current = Path(path)
+    if Path(os.path.join(path, f"{current.name}.spec")).is_file():
+        if Path(os.path.join(path, f"{current.name}.signatures.json")).is_file():
+            return path
+    parent = current.parent
+    if parent != current:
+        return find_spec_folder_with_signatures_json(f"{parent}")
+    return None
+
+def check_folder(folder):
     signatures_correct = True
+
+    # find YY (maybe ancestor of path) that has xx/YY/YY.spec
+    path = find_spec_folder_with_signatures_json(folder)
+    if path is None:
+        # no spec/signature files found in path or its ancestors
+        return signatures_correct
+
     for signature_path in Path(path).glob("*.signatures.json"):
```

Comment:
Btw. only valid signatures file is `<spec_name>.signatures.json`. In an off chance there's more than one `.signatures.json` files, all other ones will be ignored by the toolkit anyway, so it might be worth to make the check similarly strict as well.

Reply:
not a valid assumption ... adjusted accordingly

Reply:
Also, I noticed I was imprecise: there can be more `.signature.json` files when there's more `.spec` files but what I had in mind is `.signature.json` files, which don't match any `.spec` files. They may be source files for one of the `.spec` files then but I'd be very surprised to see that.